src/main/java/io/ebeaninternal/server/query/SqlTreeBuilder.java

```diff
@@ -438,7 +440,7 @@ private void addProperty(SqlTreeProperties selectProps, BeanDescriptor<?> desc,
         p = desc.findBeanProperty("id");
         selectProps.add(p);
 
-      } else if (p.isId()) {
+      } else if (p.isId() && (query == null || !query.isSingleAttribute())) {
         // do not bother to include id for normal queries as the
         // id is always added (except for subQueries)
```

Comment:
we must not skip id properties in singleAttribute queries - because it is the only one.